src/ralph/dns/views.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+import logging
+from urllib.parse import urlencode
+
+import requests
+from django.conf import settings
+from django.contrib import messages
+from django.forms import BaseFormSet, formset_factory
+from django.http import HttpResponseRedirect
+from django.utils.translation import ugettext_lazy as _
+
+from ralph.admin.views.extra import RalphDetailView
+from ralph.dns.forms import DNSRecordForm, RecordType
+
+logger = logging.getLogger(__name__)
+
+
+def get_api_kwargs(url, data=None, headers=None):
+    if headers is None:
+        headers = {}
+    headers.update({'Authorization': 'Token {}'.format(settings.DNSAAS_TOKEN)})
+
+    return {'url': url, 'data': data, 'headers': headers}
+
+
+def get_api_result(url):
+    request = requests.get(**get_api_kwargs(url))
+    json_data = request.json()
+    api_results = json_data.get('results', [])
+    if json_data.get('next', None):
+        _api_results = get_api_result(json_data['next'])
+        api_results.extend(_api_results)
+    return api_results
+
+
+def get_dns_records(ipaddresses):
+    """Gets DNS Records for `ipaddresses` by API call"""
+    dns_records = []
+    ipaddresses = [('ip', i) for i in ipaddresses]
+    url = '{}/{}/?{}'.format(
+        settings.DNSAAS_URL,
+        'api/records',
+        urlencode([
+            ('limit', 100),
+            ('offset', 0)
+        ] + ipaddresses)
+    )
+    api_results = get_api_result(url)
+    ptr_list = set([i['content'] for i in api_results if i['type'] == 'PTR'])
+
+    for item in api_results:
+        if item['type'] in {'A', 'CNAME', 'TXT'}:
+            dns_records.append({
+                'pk': item['id'],
+                'name': item['name'],
+                'type': RecordType.from_name(item['type'].lower()).id,
+                'content': item['content'],
+                'ptr': item['name'] in ptr_list
+            })
+    return dns_records
+
+
+def update_dns_records(records):
+    error = False
+    for item in records:
+        url = '/'.join([settings.DNSAAS_URL, 'api/records', str(item['pk'])])
+        data = {
+            'name': item['name'],
+            'type': RecordType.raw_from_id(int(item['type'])),
+            'content': item['content'],
+            'auto_ptr': (
+                settings.DNS_AUTO_PTR_ALWAYS if item['ptr'] and
```

Comment:
This is supported by PowerDNS
```
@receiver(post_save, sender=Record, dispatch_uid='record_create_ptr')
def create_ptr(sender, instance, **kwargs):
    if instance.auto_ptr == AutoPtrOptions.NEVER or instance.type != 'A':
        instance.delete_ptr()
        return
    instance.create_ptr()
```

create_ptr method automatically created ptr.

If user will uncheck PTR, powerdns delete ptr records, 
If already PTR powerdns refresh ptr records. 

@xliiv What do you think? 

https://github.com/allegro/django-powerdns-dnssec/blob/develop/powerdns/models/powerdns.py#L556